warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RootNegationCheckVisitor.java

```diff
@@ -44,14 +44,9 @@ private boolean hasLeafWithoutNegation(JexlNode node) {
                 child = instance.getSource();
             }
             
-            if (ASTNENode.class.equals(childClass) || ASTNotNode.class.equals(childClass) || ASTNRNode.class.equals(childClass)) {
-                // no need to inspect this path anymore
-            } else if (child == null || child.jjtGetNumChildren() == 0) {
-                // reached a leaf node without ever encountering a negation
-                foundLeaf = true;
-            } else {
-                // neither a leaf nor negation, continue looking
-                foundLeaf = hasLeafWithoutNegation(child);
+            if (!(ASTNENode.class.equals(childClass) || ASTNotNode.class.equals(childClass) || ASTNRNode.class.equals(childClass))) {
+                // either reached a leaf node without encountering a negation, or we need to continue looking
+                foundLeaf = child.jjtGetNumChildren() == 0 || hasLeafWithoutNegation(child);
```

Comment:
We don't want to keep the child null check?

Reply:
Adding the null check was an unintended functional change. Originally a null node would throw a NPE, whereas with the null check the visitor continues on.

Reply:
ok.  i figured that was one of the things sonar lint complained about.